ext/nnstreamer/tensor_filter/tensor_filter_tensorrt_v10.cc

```diff
@@ -0,0 +1,758 @@
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
+struct NvInferDeleter
+{
+    template <typename T>
+    void operator()(T* obj) const
+    {
+        delete obj;
+    }
+};
+
+template <typename T>
+using NvInferUniquePtr = std::unique_ptr<T, NvInferDeleter>;
+
+template <typename T> NvInferUniquePtr<T> makeNvInferUniquePtr (T *t)
+{
+  return NvInferUniquePtr<T>{ t };
+}
+
+struct NvInferTensorInfo {
+  const char* name;
+  nvinfer1::TensorIOMode mode;
+  nvinfer1::Dims shape;
+  nvinfer1::DataType dtype;
+  std::size_t dtype_size;
+  std::size_t volume;
+  std::size_t buffer_size;
+  void *buffer; /**< Cuda buffer */
+};
+
+/** @brief tensorrt subplugin class */
+class tensorrt_subplugin final : public tensor_filter_subplugin
+{
+  template <typename T> using UniquePtr = std::unique_ptr<T>;
+
+  public:
+  static void init_filter_tensorrt ();
+  static void fini_filter_tensorrt ();
+
+  tensorrt_subplugin ();
+  ~tensorrt_subplugin ();
+
+  tensor_filter_subplugin &getEmptyInstance ();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+
+
+  private:
+  static const char *name;
+  static const accl_hw hw_list[];
+  static const int num_hw = 0;
+  static tensorrt_subplugin *registeredRepresentation;
+
+  bool _configured{}; /**< Flag to keep track of whether this instance has been configured or not. */
+  int _device_id{}; /**< Device id of the gpu to use. */
+  gchar *_model_path{}; /**< engine file path */
+  std::filesystem::path _engine_fs_path{}; /**< filesystem path to engine file */
+  cudaStream_t _stream{}; /**< The cuda inference stream */
+
+  GstTensorsInfo _inputTensorMeta;
+  GstTensorsInfo _outputTensorMeta;
+
+  NvInferUniquePtr<nvinfer1::IRuntime> _Runtime{};
+  NvInferUniquePtr<nvinfer1::ICudaEngine> _Engine{};
+  NvInferUniquePtr<nvinfer1::IExecutionContext> _Context{};
+
+  std::vector<NvInferTensorInfo> _tensorrt_input_tensor_infos{};
+  std::vector<NvInferTensorInfo> _tensorrt_output_tensor_infos{};
+
+  void cleanup();
+  void allocBuffer (void **buffer, gsize size);
+  void loadModel (const GstTensorFilterProperties *prop);
+  void checkUnifiedMemory () const;
+  void convertTensorsInfo (const std::vector<NvInferTensorInfo>& tensorrt_tensor_infos, GstTensorsInfo &info) const;
+  std::size_t getVolume (const nvinfer1::Dims& shape) const;
+  std::size_t getTensorRTDataTypeSize (nvinfer1::DataType tensorrt_data_type) const;
+  tensor_type getNnStreamerDataType(nvinfer1::DataType tensorrt_data_type) const;
+
+  void constructNetwork(NvInferUniquePtr<nvonnxparser::IParser>& parser) const;
+  void buildSaveEngine () const;
+  void loadEngine ();
+};
+
+const char *tensorrt_subplugin::name = "tensorrt";
```

Comment:
Please make the name and its binary file name distinguishable against older tensorrt. (e.g., tensorrt10)
And, let the subplugin check and abort if the linked tensorrt version is incomaptible during the init.

We need a way to identify whether this is for tensorrt+10 or not after they are compiled and deployed although we can deprecate the older one afterwards.